### PR TITLE
Write the (pre)release workflow

### PR DESCRIPTION
**Pull request by jubianchi**
* [x] Prerelease workflow (will try to upload wheels on TestPyPI)
* [x] Release workflow
* [x] Re-enable caching

### CODE REVIEW

**Comment by jubianchi** [comment]
> * [x] Prerelease workflow (will try to upload wheels on TestPyPI)

Example build: https://github.com/jubianchi/wasmer-python/actions/runs/387544974

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -114,3 +123,85 @@ jobs:
...
+          name: wheels-${{ matrix.python }}-${{ matrix.target.id }}
+          path: target/wheels
+          if-no-files-found: error
+          retention-days: 1
+
+  prerelease:
+    name: Pre-Release
+
+    runs-on: ubuntu-18.04
+    if: startsWith(github.ref, 'refs/heads/prepare-')
```
Comment (Hywan) [inline .github/workflows/test.yml:155]: I wonder if we could trigger this with a keyword in a commit message. Like `[prerelease]`. Thoughts?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -6,58 +6,23 @@ on:
...
-          # Linux, amd64
-          { id: 'linux-amd64', os: 'ubuntu-latest', rust-target: 'x86_64-unknown-linux-gnu', rust-toolchain: 'stable' },
-          # macOS, amd64
-          { id: 'darwin-amd64', os: 'macos-latest', rust-target: 'x86_64-apple-darwin', rust-toolchain: 'stable' },
-          # Windows, amd64
-          { id: 'windows-amd64', os: 'windows-latest', rust-target: 'x86_64-pc-windows-msvc', rust-toolchain: 'stable' },
-          # Windows, x86
-          #{ id: 'windows-x86', os: 'windows-latest', rust-target: 'i686-pc-windows-msvc' },
-          # Linux, aarch64
-          #{ id: 'linux-aarch64', os: ['self-hosted', 'aarch64'], rust-target: 'aarch64-unknown-linux-gnu', rust-toolchain: 'stable' },
```
Comment (Hywan) [inline .github/workflows/release.yml:55]: Why Windows has been removed?

File: .github/workflows/test.yml (hunk 1 above)
Comment (jubianchi) [reply .github/workflows/test.yml:155]: I think we can but I also think it preferable to have the release preparation in a separate branch and force this.

File: .github/workflows/release.yml (hunk 2 above)
Comment (jubianchi) [reply .github/workflows/release.yml:55]: It has been removed from both workflows. I want to have these validated and then work on making Windows work.

**Review by Hywan — APPROVED** [review]

**Pull request merged**